src/main/java/net/fabricmc/tinyv2/V2MappingsProvider.java

```diff
@@ -0,0 +1,278 @@
+        private final Map<String, Integer> namespaceIndices;
+        private final EntryTriple[] names;
+
+        MemberEntryImpl(Map<String, Integer> namespaceIndices, EntryTriple[] names) {
+            this.namespaceIndices = namespaceIndices;
+            this.names = names;
+        }
+
+        @Override
+        public EntryTriple get(String namespace) {
+            return names[namespaceIndices.get(namespace)];
+        }
+    }
+
+    private static class ParameterEntryImpl implements MethodParameterEntry {
+        private final Map<String, Integer> namespaceIndices;
+        private final MethodParameter[] names;
+
+        ParameterEntryImpl(Map<String, Integer> namespaceIndices, MethodParameter[] names) {
+            this.namespaceIndices = namespaceIndices;
+            this.names = names;
+        }
+
+        @Override
+        public MethodParameter get(String namespace) {
+            return names[namespaceIndices.get(namespace)];
+        }
+    }
+
+    private static class LocalVariableEntryEntryImpl implements LocalVariableEntry {
+        private final Map<String, Integer> namespaceIndices;
+        private final LocalVariable[] names;
+
+        LocalVariableEntryEntryImpl(Map<String, Integer> namespaceIndices, LocalVariable[] names) {
+            this.namespaceIndices = namespaceIndices;
+            this.names = names;
+        }
+
+        @Override
+        public LocalVariable get(String namespace) {
+            return names[namespaceIndices.get(namespace)];
+        }
+    }
+
+    private static class Visitor implements TinyVisitor {
+        private Collection<ClassEntry> classEntries = new ArrayList<>();
+        private Collection<MethodEntry> methodEntries= new ArrayList<>();
+        private Collection<FieldEntry> fieldEntries= new ArrayList<>();
+        private Collection<MethodParameterEntry> methodParameterEntries= new ArrayList<>();
+        private Collection<LocalVariableEntry> localVariableEntries= new ArrayList<>();
+        private Comments comments = new CommentsImpl(new ArrayList<>(), new ArrayList<>(), new ArrayList<>()
+                , new ArrayList<>(), new ArrayList<>());
+        private List<String> namespaces;
+
+        private Map<String, Integer> namespaceIndices = new HashMap<>();
+        private int namespaceAmount;
+
+        private String[] currentClassNames;
+        private EntryTriple[] currentMemberNames;
+        private MethodParameter[] currentParameterNames;
+        private LocalVariable[] currentLocalVariableNames;
+        private List<String> currentComments;
+        private CommentType currentCommentType;
+
+        @Override
+        public void start(TinyMetadata metadata) {
+            namespaces = metadata.getNamespaces();
+            namespaceAmount = namespaces.size();
+            for (int i = 0; i < namespaceAmount; i++) {
+                namespaceIndices.put(namespaces.get(i), i);
```

Comment:
Consider replacing this map with a `ToIntFunction<String>`, which will output an `int` for a `String` input. So like
```java
private ToIntFunction<String> namespaceIndices;

// stuff

namespaceIndices = metadata::index;
```